### PR TITLE
NativeAOT-LLVM: Merge up to but not including PR67238

### PR DESCRIPTION
**Pull request by yowl**
This PR merges up to but does not include https://github.com/dotnet/runtime/pull/67238

Mostly TARGET_LOONGARCH64 changes plus removal of `Lowering::TransformUnusedIndirection` for Wasm

### CODE REVIEW

**Comment by yowl** [comment]
2 OSX failures, the Release one looks the same as mentioned in https://github.com/dotnet/runtimelab/pull/2173#issuecomment-1385832016.  The Debug one looks like an undefined reference to "_PAL_fprintf" which I'll have a look at.  Conflict resolutions can be reviewed, along with the exclusion of `TransformUnusedIndirection ` for `TARGET_WASM`

**Comment by yowl** [comment]
cc @dotnet/nativeaot-llvm

File: src/coreclr/jit/instr.cpp [hunk 1]
```diff
@@ -1589,7 +1589,6 @@ instruction CodeGen::ins_Copy(var_types dstType)
     {
         return INS_mov;
     }
-<<<<<<< HEAD
 #elif defined(TARGET_WASM)
```
Comment (SingleAccretion) [inline src/coreclr/jit/instr.cpp]: Is `instr.cpp` compiled for WASM at all? If not (which I think is the case), we should just delete all `TARGET_WASM`s from it.

File: src/coreclr/jit/lower.cpp [hunk 2]
```diff
@@ -5922,11 +5922,7 @@ GenTree* Lowering::LowerConstIntDivOrMod(GenTree* node)
             return nullptr;
         }
 
-<<<<<<< HEAD
-#if defined(TARGET_XARCH) || defined(TARGET_ARM64) || defined(TARGET_WASM) // TODO Wasm
-=======
-#if defined(TARGET_XARCH) || defined(TARGET_ARM64) || defined(TARGET_LOONGARCH64)
->>>>>>> 6543a048d7242ddf204f2e1ba0723d27c02bdfc7
+#if defined(TARGET_XARCH) || defined(TARGET_ARM64) || defined(TARGET_WASM) || defined(TARGET_LOONGARCH64) // TODO Wasm
```
Comment (SingleAccretion) [inline src/coreclr/jit/lower.cpp]: Same question as with `instr.cpp`.

File: src/coreclr/jit/unwind.cpp [hunk 3]
```diff
@@ -447,13 +447,10 @@ UNATIVE_OFFSET Compiler::unwindGetCurrentOffset(FuncInfoDsc* func)
 
 // See unwindX86.cpp
 
-<<<<<<< HEAD
 #elif defined(TARGET_WASM) // TODO
```
Comment (SingleAccretion) [inline src/coreclr/jit/unwind.cpp]: Let's change it like so:
```
#elif defined(TARGET_WASM)

// No unwind info on WASM.

```

File: src/coreclr/tools/aot/ilc.sln [hunk 4]
```diff
@@ -18,9 +18,9 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "ILCompiler.LLVM", "ILCompil
 EndProject
```
Comment (SingleAccretion) [inline src/coreclr/tools/aot/ilc.sln:18]: Solution changes - to be reverted?

**Review by SingleAccretion — COMMENTED** [review]
I would be tempted to disable OSX x64 Release in CI until we merge [upstream fixes for it](https://github.com/dotnet/runtime/pull/75421).

File: src/coreclr/tools/aot/ilc.sln (hunk 4 above)
Comment (yowl) [reply src/coreclr/tools/aot/ilc.sln:18]: yes, I don't know why VS does this all the time for me.  I'll clear my .vs folder.

Comment (yowl) [reply src/coreclr/tools/aot/ilc.sln:18]: I see, VS wants to create the configuration entries for the 2 vcxproj jit entries which are missing.  Does that not happen for you @SingleAccretion ?

Comment (SingleAccretion) [reply src/coreclr/tools/aot/ilc.sln:18]: It certainly does add things, but I always `git restore` + `Ignore` them (so not sure what they actually are).

Would be no objections from me for checking it in and seeing whether VS holds its peace from there.

File: src/coreclr/jit/lower.cpp [hunk 5]
```diff
@@ -5922,7 +5922,7 @@ GenTree* Lowering::LowerConstIntDivOrMod(GenTree* node)
             return nullptr;
         }
 
-#if defined(TARGET_XARCH) || defined(TARGET_ARM64) || defined(TARGET_WASM) || defined(TARGET_LOONGARCH64) // TODO Wasm
+#if defined(TARGET_XARCH) || defined(TARGET_ARM64) || defined(TARGET_LOONGARCH64) // TODO Wasm
```
Comment (SingleAccretion) [inline src/coreclr/jit/lower.cpp]:
```suggestion
#if defined(TARGET_XARCH) || defined(TARGET_ARM64) || defined(TARGET_LOONGARCH64)
```

**Comment by yowl** [comment]
I removed the OSX Debug build for the fprintf issue, but I will try putting it back as it could be related to https://github.com/dotnet/runtime/pull/64631

File: src/coreclr/jit/instr.cpp (hunk 1 above)
Comment (yowl) [reply src/coreclr/jit/instr.cpp]: No, its not used, have removed as suggested

File: src/coreclr/jit/lower.cpp (hunk 2 above)
Comment (yowl) [reply src/coreclr/jit/lower.cpp]: No, its not used, have removed as suggested

File: src/coreclr/jit/unwind.cpp (hunk 3 above)
Comment (yowl) [reply src/coreclr/jit/unwind.cpp]: Done, thanks

**Comment by yowl** [comment]
This is ready for final review, thanks.

**Review by SingleAccretion — APPROVED** [review]
Looks good!

**Pull request merged**